### PR TITLE
refactor: rework endpoint resolver [FRE-616]

### PR DESCRIPTION
**Pull request by grikomsn**

### CODE REVIEW

**Review by thal0x — APPROVED** [review]

File: src/config/endpoints.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  "andromeda-1": true,
+  "archway-1": true,
+  "mantle-1": true,
+  "xstaxy-1": true,
+  "axelar-dojo-1": true,
+  "laozi-mainnet": true,
+  "bitsong-2b": true,
+  "canto_7700-1": true,
+  "carbon-1": true,
+  celestia: true,
```
Comment (bpiv400) [inline src/config/endpoints.js:52]: Intentionally no quotes here?

File: src/config/endpoints.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  "crescent-1": true,
+  "cronosmainnet_25-1": true,
+  "crypto-org-mainnet": true,
+  "vota-ash": true,
+  "dydx-mainnet-1": true,
+  "dymension_1100-1": true,
+  "empowerchain-1": true,
+  "evmos_9001-2": true,
+  "fetchhub-4": true,
+  gitopia: true,
```
Comment (bpiv400) [inline src/config/endpoints.js:67]: Intentionally no quotes here?

File: src/pages/api/rest/handler.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    const endpoint = getWhitelabelEndpoint(chainID, "api") ?? chainRecord[chainID]?.apis?.rest?.[0]?.address;
+    if (!endpoint) {
+      return new Response(null, { status: 404 }); // Not Found
+    }
+
+    const headers = new Headers();
+    if (endpoint.includes("skip-api") && endpoint.endsWith(".polkachu.com")) {
+      headers.set("authorization", getPolkachuAuthHeader());
+    }
```
Comment (bpiv400) [inline src/pages/api/rest/handler.ts:28]: Its really awkward that specific logic relating to Polkachu's naming conventions leaks into the top level handler function.

File: src/pages/api/rpc/handler.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  api: {
+    externalResolver: true,
+    bodyParser: false,
+  },
+  runtime: "edge",
+};
+
+export default async function handler(req: NextRequest) {
+  try {
+    const [chainID, ...args] = req.url.split("/api/rpc/").pop()!.split("/");
```
Comment (bpiv400) [inline src/pages/api/rpc/handler.ts:18]: This logic is basically identical to the logic in the other handler, with a couple of strings changed. 

Surely we can do this more cleanly

File: src/config/endpoints.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  "lum-network-1": true,
+  "mars-1": true,
+  "migaloo-1": true,
+  "neutron-1": true,
+  "cataclysm-1": true,
+  "noble-1": true,
+  "nois-1": true,
+  "pirin-1": true,
+  "nomic-stakenet-3": true,
+  nyx: true,
```
[the remark this answers is not in the document]
Comment (grikomsn) [reply src/config/endpoints.js:84]: js can do no quotes; formatters usually prefer no quotes

File: src/pages/api/rest/handler.ts (hunk 3 above)
Comment (grikomsn) [reply src/pages/api/rest/handler.ts:28]: yeah i might change this soon, remind me tomorrow 🙏🏻

File: src/pages/api/rpc/handler.ts (hunk 4 above)
Comment (grikomsn) [reply src/pages/api/rpc/handler.ts:18]: can refactor tomorrow 🙏🏻

Comment (grikomsn) [reply src/pages/api/rpc/handler.ts:18]: https://github.com/skip-mev/ibc-dot-fun/pull/160 @bpiv400

**Pull request merged**